src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzerTests.cs

```diff
@@ -0,0 +1,694 @@
+        [InlineData("build_property.TargetFramework=net5.0-windows10.0", true)]
+        [InlineData("build_property.TargetFramework=net5.0-windows11.0\nbuild_property.TargetPlatformMinVersion=10.0;", true)]
+        [InlineData("build_property.TargetFramework=net5.0-windows10.2\nbuild_property.TargetPlatformMinVersion=10.1;", true)]
+        [InlineData("build_property.TargetFramework=net5.0-windows10.1.1.2\nbuild_property.TargetPlatformMinVersion=10.0.0.1;", true)]
+        [InlineData("build_property.TargetFramework=net5.0-windows10.2.1\nbuild_property.TargetPlatformMinVersion=9.1.1.1;", true)]
+        public async Task TfmAndTargetPlatformMinVersionWithMinimumOsAttribute(string editorConfigText, bool expectDiagnostic)
+        {
+            var invocation = expectDiagnostic ? @"[|M2()|]" : "M2()";
+            await new VerifyCS.Test
+            {
+                TestState =
+                {
+                    Sources =
+                    {
+                        $@"
+using System.Runtime.Versioning;
+
+public class Test
+{{
+    public void M1()
+    {{
+        {invocation};
+    }}
+    [MinimumOSPlatform(""Windows10.1.1.1"")]
+    public void M2()
+    {{
+    }}
+}}
+" + MockAttributesSource
+                    },
+                    AdditionalFiles = { (".editorconfig", editorConfigText) }
+                },
+                MarkupOptions = MarkupOptions.UseFirstDescriptor
+            }.RunAsync();
+        }
+
+        [Theory]
+        [InlineData("", true)]
+        [InlineData("build_property.TargetFramework=net5.0", true)]
+        [InlineData("build_property.TargetFramework=net472", false)] // TODO: because no version part it is 0.0.0.0
+        [InlineData("build_property.TargetFramework=net5.0-linux", true)]
+        [InlineData("build_property.TargetFramework=net5.0-windows", true)]
+        [InlineData("build_property.TargetFramework=net5.0-windows10.1.1.1", true)]
+        [InlineData("build_property.TargetFramework=net5.0-windows10.2", true)]
+        [InlineData("build_property.TargetFramework=net5.0-windows11.0", true)]
+        [InlineData("build_property.TargetFramework=net5.0-windows10.0", false)]
+        [InlineData("build_property.TargetFramework=net5.0-windows11.0\nbuild_property.TargetPlatformMinVersion=10.0;", false)]
+        [InlineData("build_property.TargetFramework=net5.0-windows10.2\nbuild_property.TargetPlatformMinVersion=10.1;", false)]
+        [InlineData("build_property.TargetFramework=net5.0-windows10.1.1.2\nbuild_property.TargetPlatformMinVersion=10.0.0.1;", false)]
+        [InlineData("build_property.TargetFramework=net5.0-windows10.2.1\nbuild_property.TargetPlatformMinVersion=9.1.1.1;", false)]
+        public async Task TfmAndTargetPlatformMinVersionWithObsoleteAttribute(string editorConfigText, bool expectDiagnostic)
+        {
+            var invocation = expectDiagnostic ? @"[|M2()|]" : "M2()";
+            await new VerifyCS.Test
```

Comment:
I noticed you consume a similar `VerifyCS.Test` object in more than one place. Consider moving its code to its own method.

For example, [I did this in one of my analyzer tests ](https://github.com/dotnet/roslyn-analyzers/blob/d7529293a04f71e007f1a7b74cba6378979589f3/src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocationsTests.cs#L4221-L4232) when I needed many tests to run with a minimum version of C# 8.